module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,315 @@
+                                    "$project": {
+                                        '_id': "$pollId",
+                                        'name': '$pollName'
+                                    }
+                                },
+                                { "$sort": { createdAt: 1 } }
+                            ])
+                ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+
+                        if (Array.isArray(response[0].data) && response[0].data.length > 0) {
+                            result = result.concat(response[0].data);
+                        }
+
+                        if (response[1].length > 0) {
+                            result = result.concat(response[1]);
+                        }
+                    });
+
+                let uniquePolls = [];
+
+                if (result.length > 0) {
+
+                    let mymap = new Map();
+                    uniquePolls = result.filter(el => {
+                        const val = mymap.get(el.name);
+                        if (val) {
+                            if (el._id < val) {
+                                mymap.delete(el.name);
+                                mymap.set(el.name, el._id);
+                                return true;
+                            } else {
+                                return false;
+                            }
+                        }
+                        mymap.set(el.name, el._id);
+                        return true;
+                    });
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: uniquePolls
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    /**
+    * Validate poll submission.
+    * @method
+    * @name validatePollSubmission
+    * @param {String} pollId - pollId
+    * @param {String} userId - userId
+    * @returns {String} - message.
+    */
+
+   static validatePollSubmission(pollId= "", userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            if (pollId == "") {
+                throw new Error(messageConstants.apiResponses.POLL_ID_REQUIRED_CHECK)
+            }
+
+            if (userId == "") {
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+            }
+
+            let pollDocument = await pollsHelper.validatePoll
+            (
+                pollId
+            )
+            
+            if (!pollDocument.success && !pollDocument.data) {
+               return resolve(pollDocument);
```

Comment:
@deepakr123 - Why a different resolve here ?

From this function - resolve has to be as below 

success: true,
message: messageConstants.apiResponses.POLL_SUBMISSION_VALIDATED,
data: true // True if submission can be made else false